### PR TITLE
Add combobox selection for VCF to validate against

### LINKED ISSUE #520: Select validation VCF in /examine
Allow selecting of any VCF (within a given project) to be compared as a validation VCF against the selected /examine VCF.

Especially necessary as of #510 when we can no longer say that a VCF has a validation VCF.

### PR DESCRIPTION
**Pull request by ihodes**
Also start the slow process of renaming run -> vcf.

pdiff tests.yaml is updated, and works in Chrome (and is purportedly supported in PhantomJS?) but isn't generating my expected screenshot. Mind giving it a try?

fixes #520 

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/cycledash/538)

### CODE REVIEW

File: tests/data.sql [hunk 1]
```diff
@@ -71,6 +72,11 @@ COPY genotypes (vcf_id, sample_name, contig, "position", id, reference, alternat
...
 1	TUMOR	20	75254	\N	C	A	\N	\N	DEFB125	\N	\N	\N	\N	\N	\N	\N	74	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	0/1	40	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	1	9	7.9203e-12	0.11567	20	20	50%	[5, 15, 14, 6]	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N
 1	NORMAL	20	75254	\N	C	A	\N	\N	DEFB125	\N	\N	\N	\N	\N	\N	\N	74	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	0/1	34	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	1	9	7.9203e-12	0.11567	22	11	33.33%	[13, 9, 5, 6]	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N
+2	TUMOR	20	66370	\N	G	A	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	66	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	1/1	34	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	1	0	2.6498e-39	1.0	0	34	100%	[0, 0, 15, 19]	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N	\N
```
Comment (ihodes) [inline tests/data.sql]: Hand-written postgres dump!

**Comment by danvk** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/cycledash/538)



---

Generally looks good -- a couple comments for you.

---

<sup>**[cycledash/genotypes.py, line 87 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJhshuaZisy9j48AIG)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/genotypes.py#L87)):</sup>
`selectedVcfId` → `validateVcfId`?

---

<sup>**[cycledash/static/js/examine/components/ExamineInformation.js, line 15 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJieC01XVSE-Km2pNO)** ([raw file](https://github.com/hammerlab/cycledash/blob/b98b9b38a8a4ab3e28b18913c0cbe949039dba6c/cycledash/static/js/examine/components/ExamineInformation.js#L15)):</sup>
no longer relevant?

---

<sup>**[cycledash/static/js/examine/components/ExaminePage.js, line 20 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJis34NY74iD7r02PT)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/static/js/examine/components/ExaminePage.js#L20)):</sup>
What's the relationship between `vcf` and `vcfs`? Do they have the same shape? Is `vcf` ∈ `vcfs`? (They should be next to each other in the `propTypes` list & have a comment explaining.)

---

<sup>**[cycledash/static/js/examine/components/ExaminePage.js, line 20 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJk6lUAuLF6sApdS73)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/static/js/examine/components/ExaminePage.js#L20)):</sup>
This would be clearer as `vcfsForValidation`

---

<sup>**[cycledash/static/js/examine/components/VCFValidation.js, line 1 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJj9sK-cmJXYzhv92k)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/static/js/examine/components/VCFValidation.js#L1)):</sup>
Add a one-liner explaining what this module is.

---

<sup>**[cycledash/static/js/examine/components/VCFValidation.js, line 16 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJjgTJmZUQdQvjDv96)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/static/js/examine/components/VCFValidation.js#L16)):</sup>
the line break looks kinda odd here -- put `this.props.vcfs` on the previous line and the `.map` on its own line? or merged with the next one, i.e. `.map(v => ...)` on its own line?

---

<sup>**[cycledash/static/js/examine/components/VCFValidation.js, line 20 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJjx65Et08VZnokbE_)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/static/js/examine/components/VCFValidation.js#L20)):</sup>
High level question: do we want this to be "validate against" or the more generic "compare against"?

---

<sup>**[cycledash/static/js/examine/RecordStore.js, line 95 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJkGucLRB42qNkhc7p)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/static/js/examine/RecordStore.js#L95)):</sup>
Use `Number(foo)` or do `parseInt(foo, 10)`.

---

**Comment by ihodes** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/cycledash/538)







---

<sup>**[cycledash/genotypes.py, line 87 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJhshuaZisy9j48AIG)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/genotypes.py#L87)):</sup>
`compare_to_vcf_id` work? 

---

<sup>**[cycledash/static/js/examine/components/ExamineInformation.js, line 15 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJieC01XVSE-Km2pNO)** ([raw file](https://github.com/hammerlab/cycledash/blob/b98b9b38a8a4ab3e28b18913c0cbe949039dba6c/cycledash/static/js/examine/components/ExamineInformation.js#L15)):</sup>
:( reviewable http://cl.ly/image/0f3f3q2p2C0i

Yeah, we don't need the ID up front.1

---

<sup>**[cycledash/static/js/examine/components/VCFValidation.js, line 20 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJjx65Et08VZnokbE_)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/static/js/examine/components/VCFValidation.js#L20)):</sup>
"Compare to" sound good?

---

**Comment by danvk** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/cycledash/538)







---

<sup>**[cycledash/genotypes.py, line 87 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJhshuaZisy9j48AIG)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/genotypes.py#L87)):</sup>
I was referring to the URL parameter, actually—`compareToVcfId` would make more sense than `selectedVcfId`.

---

<sup>**[cycledash/static/js/examine/components/VCFValidation.js, line 22 \[r1\]](https://reviewable.io:443/reviews/hammerlab/cycledash/538#-JkJjx65Et08VZnokbE_)** ([raw file](https://github.com/hammerlab/cycledash/blob/a9a04241bfba27f41f1e6f5db8161b55909ae903/cycledash/static/js/examine/components/VCFValidation.js#L22)):</sup>
Sure -- we should get rid of "validation" elsewhere in this change, then, e.g. `handleComparisonVcfChange`, `vcf-comparisons`, etc.

Or leave it as "Validate against:" here. I'm fine either way, I just want a single term for it.

---

**Comment by ihodes** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/cycledash/538)







---

**Comment by ihodes** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/cycledash/538)







---

**Comment by ihodes** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/cycledash/538)







---

**Comment by danvk** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/cycledash/538)



---

LGTM. Need me to update screenshots?

---

**Comment by danvk** [comment]
I updated the screenshots & pushed the commit to your branch.

**Comment by danvk** [comment]
(looks like there may be some CSS issues with the dropdown?)

**Comment by ihodes** [comment]
Hmm the validation isn't occurring for you either. Note no check marks on the validated VCF screenshot. CSS/drop down styling issue only occurs in phantomjs, not sure why :(

**Pull request merged**